flang/lib/Semantics/resolve-directives.cpp

```diff
@@ -1507,12 +1525,13 @@ void OmpAttributeVisitor::ResolveOmpName(
         AddToContextObjectWithDSA(*resolvedSymbol, ompFlag);
       }
     }
-  } else if (ompFlag == Symbol::Flag::OmpCriticalLock) {
-    // Create a new symbol.
+  } else if (ompFlagsRequireNewSymbol.test(ompFlag)) {
     const auto pair{GetContext().scope.try_emplace(
-        name.source, Attrs{}, UnknownDetails{})};
+        name.source, Attrs{}, ObjectEntityDetails{})};
```

Comment:
If you set it back to `UnknownDetails` it will work. I think for ObjectEntityDetails there are additional requirements.

Reply:
Thanks. This solved the erring test case.